### DISCUSSION TITLE
Is it possible to use the sidebar or searchbar in a template via a variable?

### DISCUSSION THREAD

**Discussion by astroDimitrios** [quarto-dev/quarto-cli#10929] [Q&A] [answered]
### Description

In the [toc partial](https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/html/pandoc/toc.html) you add the table of contents for the page via the `$table-of-contents$` variable.

Is something similar possible at all for the search box/search functionality or the sidebar nav? I am trying to convert some html templates to quarto and currently we don't have search functionality but have a space for where it should be, and the sidebar is generated by some extra R.

**Comment by mcanouil** [comment, marked as answer]
No, the website envelope is handled directly in TypeScript and with various options described in the documentation for website/book projects

- https://quarto.org/docs/websites/website-navigation.html

(Because, it's HTML, you can use (S)CSS and/or JavaScript)
